nixos/lib/eval-config-system.nix

```diff
@@ -0,0 +1,56 @@
+# DO NOT IMPORT. Use nixpkgsFlake.lib.nixos, or import (nixpkgs + "/nixos/lib")
+{ lib }: # read -^
+
+let
+
+  # Base: Use the miminal module evaluation
+  evalModulesMinimal = (import ./default.nix {
+    inherit lib;
+    # Implicit use of feature is noted in implementation.
+    featureFlags.minimalModules = { };
+  }).evalModules;
+
+  baseModules = import ../modules/module-list.nix;
+
+  # See nixos/doc/manual/development/lib-modules-eval.section.md
+  # or https://nixos.org/manual/nixos/unstable/index.html#sec-modules-evaluation
+  # NOTE: We keep this interface simple, because almost everything can and should be done through `modules`!
+  evalSystemConfiguration = args@{ prefix ? [], modules ? [], specialArgs ? {} }: let
+
+    # Locate Modules: Ensure custom modules are located to guarantee a better user experience.
+    #                 Non-located modules are inline modules and ones imported from value instead
+    #                 of a file reference from which the importer could infer the location.
+    #                 For example when idiomatically consuming a public interface via
+    #                 `flake.nixosModules.my-module` instead of a private one via - say -
+    #                 `(flake + /modules/my-module.nix)`.
+    # Note: `setDefaultModuleLocation` only sets a location, if not set. So if you want to configure
+    #       this yourself, just ensure that all modules in the `modules` list already have one set.
+    modulesLocation = (builtins.unsafeGetAttrPos "modules" args).file or null;
+    locatedModules = if modulesLocation == null then modules
+      else map (lib.setDefaultModuleLocation modulesLocation) modules;
+
+    # `noUserModules` is a contract consumed by
+    # modules/system/activation/specialisation.nix
+    # TODO (during review): critically revise & discuss! (and then remove this line :-)
+    noUserModules = standardModules;
```

Comment:
Is that the right name for the contract at the site of consuption?

```nix
{
          extend =
            if local.config.inheritParentConfig
            then extendModules
            else noUserModules.extendModules;
}
``` 

Imho, "no user modules" is a bit ambigous as it refers to "everything else" and that only makes sense for someone who knows what "everything" is.

`standardModules` might not be perfect, but a positive rather than a negative naming might help understanding.

Overarching though is the question if we can change that contract (here or in a follow up PR), at all?

Reply:
It's not great, and I'm afraid I'm the one who's responsible for this name, as I've cleaned up specialisations.
Let's keep the name as is, certainly for now, and not introduce new names if we don't have to. The PR is scary enough as it is.

What we can do though, is document the status quo.

Reply:
Ok, that sounds good. Removing the code the comment and resolving this question.